Add databuilder Docs (#596)
opensafely/documentation: INSTALL.md

reference to them in
[docs/study-def-variables.md](./docs/study-def-variables.md).

### Updating Data Builder Definitions

Data Builder is a documentation dependency which we can't install directly.
CloudFlare Pages' latest Python version is 3.7, but Data Builder is using 3.9 features.

> Rather than limit couple Data Builder to the versions supported by CloudFlare Pages we've opted to have it generate a JSON file which we commit to this repo.

Reviewer: nit:

```suggestion
Rather than limit Data Builder to the versions supported by Cloudflare Pages, we've opted to generate a JSON file from Data Builder and commit that JSON to this repo.
```

Author: I'm always mixing up the capitalisation because of CloudFront 🤦‍♂️

Author: Done and rebased in.
